_posts/2022-03-10-workflow.md

```diff
@@ -0,0 +1,33 @@
+---
+layout: post
+title:  "Workflow"
+date:   2022-03-18
+author: Nandi Moksnes
+category: Open Science
+tags: workflow, dataset, data, software, teaching kit, teaching material
+---
+
+<div style="text-align: justify">
+ 
+## Workflow
+
+A computational workflow combines data and it's opearations of the data. Several processes are coordinated to handel dependencies between different computational steps. Workflows looks at methods which are used for (some or all of) the following: collecting data, preparing, analysis and predictive modelling and simulation of data ([Gobel et al (2020)][1]). A tool or a model can be concidered a computational  workflow as they can support the analysis.
+ 
+![Workflow](https://user-images.githubusercontent.com/30128518/157625071-57a8a629-8319-4168-bc57-f92c4851b2ba.png)
+
+Creating a workflow can be divided into three stages.
+- First, you need to identify and describe the process and steps needed to characterise the relationship between the data   inputs and outputs. Mapping the required formats and data for potential project based database or visualisation is important to reach the desired final product.
+ - Second, when developing the workflow the scripts and codes that are produced needs to be easily found and also clearly marked which sequence they are to be executed. Here several options are possible to support, where in computational workflows [snakemake](https://snakemake.readthedocs.io/en/stable/) is one option which executes the required steps through a Snakefile until the final output is obtained. If the computations are in Python then a [Jupyter Notebook](https://jupyter.org/) (which allows for markdown comments) can be easy to describe the steps, execute functions and plot some of the results.
+ - Third, it is important to clearly defined the different steps so that it can easily be implmened with tests and checking that the workflow is working properly. This will help to identify bugs and issues that could arise.
```

Comment:
```suggestion
3.  It is important to clearly define the different steps that need to be executed, so that the workflow can be easily implemented with tests to check that it is working properly. This will help to identify bugs and issues that could arise later on.

```